supporting_crates/u256/src/lib.rs

```diff
@@ -0,0 +1,367 @@
+            x2 -= &y2;
+
+            prop_assert_eq!(x1.as_limbs(), x2.as_limbs());
+        });
+
+        proptest!(|(x_limbs: [u64; 4], y_limbs: [u64; 4])| {
+            let (mut x1, mut x2) = from_limbs(x_limbs);
+            let (y1, y2) = from_limbs(y_limbs);
+
+            x1 ^= &y1;
+            x2 ^= &y2;
+
+            prop_assert_eq!(x1.as_limbs(), x2.as_limbs());
+        });
+
+        proptest!(|(x_limbs: [u64; 4], y_limbs: [u64; 4])| {
+            let (mut x1, mut x2) = from_limbs(x_limbs);
+            let (y1, y2) = from_limbs(y_limbs);
+
+            x1 &= &y1;
+            x2 &= &y2;
+
+            prop_assert_eq!(x1.as_limbs(), x2.as_limbs());
+        });
+
+        proptest!(|(x_limbs: [u64; 4], y_limbs: [u64; 4])| {
+            let (mut x1, mut x2) = from_limbs(x_limbs);
+            let (y1, y2) = from_limbs(y_limbs);
+
+            x1 |= &y1;
+            x2 |= &y2;
+
+            prop_assert_eq!(x1.as_limbs(), x2.as_limbs());
+        });
+
+        proptest!(|(x_limbs: [u64; 4], s: u32)| {
+            let (mut x1, mut x2) = from_limbs(x_limbs);
+
+            x1 >>= s;
+            x2 >>= s;
+
+            prop_assert_eq!(x1.as_limbs(), x2.as_limbs());
+        });
+
+        proptest!(|(x_limbs: [u64; 4], s: u32)| {
+            let (mut x1, mut x2) = from_limbs(x_limbs);
+
+            x1 <<= s;
+            x2 <<= s;
+
+            prop_assert_eq!(x1.as_limbs(), x2.as_limbs());
+        });
+
+        proptest!(|(x_limbs: [u64; 4])| {
+            let (mut x1, mut x2) = from_limbs(x_limbs);
+
+            x1.not_mut();
+            x2.not_mut();
+
+            prop_assert_eq!(x1.as_limbs(), x2.as_limbs());
+        })
+    }
+
+    #[test]
+    fn compare_bytes() {
+        proptest!(|(bytes: [u8; 32])| {
+            let bytes1 = naive::U256::from_be_bytes(&bytes).to_be_bytes();
+            let bytes2 = risc_v::U256::from_be_bytes(&bytes).to_be_bytes();
+
+            prop_assert_eq!(bytes1, bytes);
+            prop_assert_eq!(bytes1, bytes2);
+        });
+
+        proptest!(|(bytes: [u8; 32])| {
+            let bytes1 = naive::U256::from_le_bytes(&bytes).to_le_bytes();
+            let bytes2 = risc_v::U256::from_le_bytes(&bytes).to_le_bytes();
+
+            prop_assert_eq!(bytes1, bytes);
+            prop_assert_eq!(bytes1, bytes2);
+        });
+
+        proptest!(|(bytes: [u8; 32])| {
+            let mut x1 = naive::U256::from_le_bytes(&bytes);
+            let mut x2 = risc_v::U256::from_le_bytes(&bytes);
+
+            x1.bytereverse();
+            x2.bytereverse();
+
+            let bytes1 = x1.to_be_bytes();
+            let bytes2 = x2.to_be_bytes();
+
+            prop_assert_eq!(bytes1, bytes);
+            prop_assert_eq!(bytes1, bytes2);
+        });
+
+        proptest!(|(bytes: [u8; 32], byte_idx: usize, bit_idx: usize)| {
+            let x1 = naive::U256::from_le_bytes(&bytes);
+            let x2 = risc_v::U256::from_le_bytes(&bytes);
+
+            let byte_idx = byte_idx % 32;
```

Comment:
Would be nice to actually check panics for corner cases. But this is optional, we can skip it